pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -19,33 +19,243 @@ package dockershim
+	createResp, err := ds.client.CreateContainer(*createConfig)
+	if err != nil || createResp == nil {
+		return "", fmt.Errorf("failed to create a sandbox for pod %q: %v", config.GetName(), err)
+	}
+
+	// Step 3: Start the sandbox container.
+	// Assume kubelet's garbage collector would remove the sandbox later, if
+	// startContainer failed.
+	err = ds.StartContainer(createResp.ID)
+	return createResp.ID, err
 }
 
 // StopPodSandbox stops the sandbox. If there are any running containers in the
 // sandbox, they should be force terminated.
 func (ds *dockerService) StopPodSandbox(podSandboxID string) error {
-	return fmt.Errorf("Not implemented")
+	return ds.client.StopContainer(podSandboxID, defaultSandboxGracePeriod)
 }
 
 // DeletePodSandbox deletes the sandbox. If there are running containers in the
 // sandbox, they should be forcibly deleted.
 func (ds *dockerService) DeletePodSandbox(podSandboxID string) error {
-	return fmt.Errorf("Not implemented")
+	return ds.client.RemoveContainer(podSandboxID, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
 }
 
-// PodSandboxStatus returns the Status of the PodSandbox.
+// PodSandboxStatus returns the status of the PodSandbox.
 func (ds *dockerService) PodSandboxStatus(podSandboxID string) (*runtimeApi.PodSandboxStatus, error) {
-	return nil, fmt.Errorf("Not implemented")
+	// Inspect the container.
+	r, err := ds.client.InspectContainer(podSandboxID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse the timstamps.
+	createdAt, _, _, err := getContainerTimestamps(r)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse timestamp for container %q: %v", podSandboxID, err)
+	}
+	ct := createdAt.Unix()
+
+	// Translate container to sandbox state.
+	state := runtimeApi.PodSandBoxState_NOTREADY
+	if r.State.Running {
+		state = runtimeApi.PodSandBoxState_READY
+	}
+
+	// TODO: We can't really get the IP address from the network plugin, which
+	// is handled by kubelet as of now. Should we amend the interface? How is
+	// this handled in the new remote runtime integration?
+	// See DockerManager.determineContainerIP() for more details.
+	// For now, just assume that there is no network plugin.
+	// Related issue: https://github.com/kubernetes/kubernetes/issues/28667
+	var IP string
+	if r.NetworkSettings != nil {
+		IP = r.NetworkSettings.IPAddress
+		// Fall back to IPv6 address if no IPv4 address is present
+		if IP == "" {
+			IP = r.NetworkSettings.GlobalIPv6Address
+		}
+	}
+	network := &runtimeApi.PodSandboxNetworkStatus{Ip: &IP}
+	netNS := getNetworkNamespace(r)
+
+	return &runtimeApi.PodSandboxStatus{
+		Id:        &r.ID,
+		Name:      &r.Name,
+		State:     &state,
+		CreatedAt: &ct,
+		// TODO: We write annotations as labels on the docker containers. All
+		// these annotations will be read back as labels. Need to fix this.
+		Labels:  r.Config.Labels,
+		Network: network,
+		Linux:   &runtimeApi.LinuxPodSandboxStatus{Namespaces: &runtimeApi.Namespace{Network: &netNS}},
```

Comment:
This one: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/api/v1alpha1/runtime/api.pb.go#L768